xarray/namedarray/core.py

```diff
@@ -855,6 +861,249 @@ def _to_dense(self) -> NamedArray[Any, _DType_co]:
         else:
             raise TypeError("self.data is not a sparse array")
 
+    def permute_dims(
+        self,
+        *dim: _DimsLike | ellipsis,
+        missing_dims: ErrorOptionsWithWarn = "raise",
+    ) -> NamedArray[Any, _DType_co]:
+        """Return a new object with transposed dimensions.
+
+        Parameters
+        ----------
+        *dim : Hashable, optional
+            By default, reverse the order of the dimensions. Otherwise, reorder the
+            dimensions to this order.
+        missing_dims : {"raise", "warn", "ignore"}, default: "raise"
+            What to do if dimensions that should be selected from are not present in the
+            NamedArray:
+            - "raise": raise an exception
+            - "warn": raise a warning, and ignore the missing dimensions
+            - "ignore": ignore the missing dimensions
+
+        Returns
+        -------
+        NamedArray
+            The returned NamedArray has permuted dimensions and data with the
+            same attributes as the original.
+
+
+        See Also
+        --------
+        numpy.transpose
+        """
+
+        if not dim:
+            dims = self.dims[::-1]
+        else:
+            dims = tuple(infix_dims(dim, self.dims, missing_dims))
+
+        if len(dims) < 2 or dims == self.dims:
+            # no need to transpose if only one dimension
+            # or dims are in same order
+            return self.copy(deep=False)
+
+        axes = self.get_axis_num(dims)
+        data = self._data.transpose(
+            axes
+        )  # TODO: replace this with use array-api-compat function
+        return self._replace(dims=dims, data=data)
+
+    @property
+    def T(self) -> NamedArray[Any, _DType_co]:
+        """Return a new object with transposed dimensions."""
+        if self.ndim != 2:
+            raise ValueError(
+                f"x.T requires x to have 2 dimensions, got {self.ndim}. Use x.permute_dims() to permute dimensions."
+            )
+
+        data = self._data[::-1, ::-1]
```

Comment:
```suggestion
        data = self._data.T
```

This isn't a transpose!